include/swift/AST/Decl.h

```diff
@@ -4788,8 +4788,12 @@ enum class PropertyWrapperSynthesizedPropertyKind {
 /// VarDecl - 'var' and 'let' declarations.
 class VarDecl : public AbstractStorageDecl {
   friend class NamingPatternRequest;
+  friend class GetFieldIndexRequest;
+  friend class ComputeFieldIndicesRequest;
   NamedPattern *NamingPattern = nullptr;
 
+  int fieldIndex = -1;
```

Comment:
It's unfortunate that we'd still need to cache index in the decl but it would be expensive to compute indexes with request for an individual property since there could be quite a few stored properties...